src/slider/index.ts

```diff
@@ -0,0 +1,140 @@
+    const handleUpper = this.state.handleUpper;
+    const handleLower = this.state.handleLower;
+    const handleLowerLeft = handleLower.getBoundingClientRect().left;
+    const handleUpperRight = handleUpper.getBoundingClientRect().right;
+    const mouseLeft = event.clientX;
+    if (Math.abs(mouseLeft - handleLowerLeft) < Math.abs(mouseLeft - handleUpperRight)) {
+      this.setup(handleLower);
+    } else {
+      this.setup(handleUpper);
+    }
+    this.state = { ...this.state, mouseLeft };
+    this.moveElements();
+  }
+
+  onMouseDown = (event: MouseEvent & Slider.Event) => {
+    event.preventDefault();
+    event.stopPropagation();
+    this.setup(event.target);
+    utils.WINDOW().document.addEventListener('mousemove', this.onMouseMove);
+    utils.WINDOW().document.addEventListener('touchmove', this.onTouchMove);
+    utils.WINDOW().document.addEventListener('mouseup', this.onMouseUp);
+    utils.WINDOW().document.addEventListener('touchend', this.onMouseUp);
+  }
+
+  onMouseMove = (event: MouseEvent & Slider.Event) => {
+    this.state = { ...this.state, mouseLeft: event.clientX };
+    this.moveElements();
+  }
+
+  onTouchMove = (event: TouchEvent & Slider.Event) => {
+    const touchEvent = event.changedTouches[0];
+    if (touchEvent.target === this.state.handle) {
+      this.state = { ...this.state, mouseLeft: touchEvent.clientX };
+      this.moveElements();
+    }
+  }
+
+  onMouseUp = (event: MouseEvent & Slider.Event) => {
+    utils.WINDOW().document.removeEventListener('mousemove', this.onMouseMove);
+    utils.WINDOW().document.removeEventListener('touchmove', this.onTouchMove);
+    utils.WINDOW().document.removeEventListener('mouseup', this.onMouseUp);
+    utils.WINDOW().document.removeEventListener('touchend', this.onMouseUp);
+  }
+
+  setup = (handle: Element) => {
+    const baseDimensions = this.root.querySelector('.gb-slider__base').getBoundingClientRect();
+    const handleLower = this.state.handleLower;
+    const handleUpper = this.state.handleUpper;
+    const isLower = handleLower === handle;
+    // tslint:disable-next-line max-line-length
+    const limit = parseInt( (isLower ? handleUpper : handleLower).style.left, 10);
+    this.state = {
+      ...this.state,
+      baseLength: baseDimensions.width,
+      baseLeft: baseDimensions.left,
+      handle,
+      isLower,
+      handleLeft: handle.getBoundingClientRect().left,
+      limit
+    };
+  }
+
+  moveElements() {
+    const percentage = this.calcMouseToPercentage();
+    const overlap = percentage === this.state.limit;
+    this.state.handle.style.left = percentage + '%';
+    this.moveConnect(percentage);
+    if (overlap && this.state.isLower) {
+      this.state.handleLower.style['z-index'] = 10;
+    } else {
+      this.state.handleLower.style['z-index'] = 1;
+    }
+  }
+
+  calcMouseToPercentage() {
+    const location = this.state.mouseLeft - this.state.baseLeft;
+    let percentage = (location / this.state.baseLength) * 100;
```

Comment:
can be a `const`, and we don't need parentheses around `location / this.state.baseLength`